aci/resource_aci_interface_config.go

```diff
@@ -0,0 +1,420 @@
+			// Card ID validation
+			if card, err = strconv.Atoi(interfaceParts[0]); err != nil || !InBetween(card, 1, 64) {
+				errors = append(errors, map[string]string{
+					"Summary": fmt.Sprintf("Card ID: %v is invalid", interfaceParts[0]),
+					"Detail":  "Card ID must be in the range of 1 to 64.",
+				})
+			}
+			// Port ID validation
+			if port, err = strconv.Atoi(interfaceParts[1]); err != nil || !InBetween(port, 1, 128) {
+				errors = append(errors, map[string]string{
+					"Summary": fmt.Sprintf("Port ID: %v is invalid", interfaceParts[1]),
+					"Detail":  "Port ID must be in the range of 1 to 128.",
+				})
+			}
+			// Sub Port ID validation
+			if len(interfaceParts) == 3 {
+				if subPort, err = strconv.Atoi(interfaceParts[2]); err != nil || !InBetween(subPort, 0, 16) {
+					errors = append(errors, map[string]string{
+						"Summary": fmt.Sprintf("Sub Port ID: %v is invalid", interfaceParts[2]),
+						"Detail":  "Sub Port ID must be in the range of 0 to 16.",
+					})
+				}
+			}
+		} else {
+			// If the interfaceParts length is less than 2 or greater than 3, it returns error.
+			errors = append(errors, invalidInterfaceError)
+		}
+	} else {
+		// If not interface, it returns error.
+		errors = append(errors, invalidInterfaceError)
+	}
+	for _, error := range errors {
+		diags = append(diags, diag.Diagnostic{
+			Severity:      diag.Error,
+			Summary:       error["Summary"],
+			Detail:        error["Detail"],
+			AttributePath: path,
+		})
+	}
+	return diags
+}
+
+func getAndSetRemoteInterfaceConfiguration(dn string, client *client.Client, d *schema.ResourceData) error {
+	interfaceConfigCont, err := client.Get(dn)
+	if err != nil {
+		return err
+	}
+
+	var interfaceConfigMap map[string]string
+	portType := d.Get("port_type").(string)
+
+	if portType == "access" {
+		interfaceConfig := models.InfraPortConfigurationFromContainer(interfaceConfigCont)
+		interfaceConfigMap, err = interfaceConfig.ToMap()
+
+		// To match the terraform plan stage changes with config file
+		if interfaceConfigMap["brkoutMap"] == "none" && d.Get("breakout").(string) == "" {
+			d.Set("breakout", nil)
+		} else {
+			d.Set("breakout", interfaceConfigMap["brkoutMap"])
+		}
+
+		d.Set("pc_member", interfaceConfigMap["pcMember"])
+	} else if portType == "fabric" {
+		interfaceConfig := models.FabricPortConfigurationFromContainer(interfaceConfigCont)
+		interfaceConfigMap, err = interfaceConfig.ToMap()
+	}
+
+	if err != nil {
+		return err
+	}
+
+	if interfaceConfigMap["dn"] == "" {
+		d.Set("breakout", nil)
+		d.Set("pc_member", nil)
+		return fmt.Errorf("Interface Configuration: %s not found", dn)
+	}
+
+	if nodeId, err := strconv.Atoi(interfaceConfigMap["node"]); err == nil {
+		d.Set("node", nodeId)
+	} else {
+		return err
+	}
+
+	d.Set("annotation", interfaceConfigMap["annotation"])
+	d.Set("interface", fmt.Sprintf("%s/%s/%s", interfaceConfigMap["card"], interfaceConfigMap["port"], interfaceConfigMap["subPort"]))
```

Comment:
Does this means that you always save it as x/y/z even when there is no subPort?

Reply:
Yes, If there is not valid subPort then we get 0 as a default subPort value.